src/Phoenix/PhoenixClient.ts

```diff
@@ -167,6 +167,9 @@ export class PhoenixClient {
       });
       return response.status === HttpStatusCodes.OK;
     } catch (error) {
+      TelemetryProcessor.traceFailure(Action.PhoenixDBAccountAllowed, {
```

Comment:
It'll be good to use `traceStart` and `traceFailure/traceSuccess` to log the complete funnel